extensions/ql-vscode/src/queries.ts

```diff
@@ -48,25 +48,25 @@ export class QueryInfo {
    */
   sortedResultsInfo: Map<string, SortedResultSetInfo>;
   dataset: vscode.Uri; // guarantee the existence of a well-defined dataset dir at this point
-
+  queryId: number;
   constructor(
     public program: messages.QlProgram,
     public dbItem: DatabaseItem,
     public queryDbscheme: string, // the dbscheme file the query expects, based on library path resolution
     public quickEvalPosition?: messages.Position,
     public metadata?: cli.QueryMetadata,
   ) {
-    this.compiledQueryPath = path.join(tmpDir.name, `compiledQuery${queryCounter}.qlo`);
+    this.queryId = queryCounter++;
```

Comment:
Could potentially rename `queryCounter` to `nextQueryId` to make the distinction between these variables clearer.